### DISCUSSION TITLE
can't use real time data

### DISCUSSION THREAD

**Discussion by bobwatcherx** [pocketbase/pocketbase#233] [Q&A]
how to use realtime data

I studied the documentation
but it didn't work
and got an error in the browser

The connection to http://localhost:8090/api/realtime was interrupted while the page was loading.

I've curled
with like this
`curl -N http://localhost:8090/api/realtime`

just appear
like this
```
id:7A0KyejknPVgCJqFUOPV71NU2LDHzD0cDFYMECj0
event:PB_CONNECT
data:{"clientId":"7A0KyejknPVgCJqFUOPV71NU2LDHzD0cDFYMECj0"}
```

this my code :
```
const source = new EventSource("http://localhost:8090/api/realtime")
source.open = e => console.log("terbuka",e)
source.onmessage = e => console.log("pesan" + e);
source.addEventListener("PB_CONNECT",e=>{
  console.log("evennya" + JSON.stringify(e))
    client.Realtime.subscribe("testapp", function (e) {
    console.log(e.record)
})

})
```
when tell me what's wrong and how to use it ? Thank you

**Comment by ganigeorgiev** [comment]
Why do you create the EventSource object on your own? `client.Realtime.subcribe()` will take care for that.

You can check the examples from the Realtime docs - https://pocketbase.io/docs/api-realtime/.
Or in other words, the following should be enough:

```js
import PocketBase from 'pocketbase';

const client = new PocketBase('http://localhost:8090');

// (Optionally) authenticate
client.Users.authViaEmail("test@example.com", "123456");

// Subscribe to changes in any record from "YOUR_COLLECTION"
client.Realtime.subscribe("YOUR_COLLECTION", function (e) {
    console.log(e.record);
});
```

**Reply by bobwatcherx** [reply]
> Why do you create the EventSource object on your own? `client.Realtime.subcribe()` will take care for that.
> 
> You can check the examples from the Realtime docs - https://pocketbase.io/docs/api-realtime/. Or in other words, the following should be enough:
> 
> ```js
> import PocketBase from 'pocketbase';
> 
> const client = new PocketBase('http://localhost:8090');
> 
> // (Optionally) authenticate
> client.Users.authViaEmail("test@example.com", "123456");
> 
> // Subscribe to changes in any record from "YOUR_COLLECTION"
> client.Realtime.subscribe("YOUR_COLLECTION", function (e) {
>     console.log(e.record);
> });
> ```

![jad2](https://user-images.githubusercontent.com/52072811/181177531-4d7f3de7-101d-4617-92a3-d37f820b7a50.gif)
in my console terminal :
```
2022/07/27 13:30:55 Realtime connection closed (cancelled request): WnFEDe0j7EObNtYWub7XqP5RAFaGAs7ztOFEkBci
[0.00ms] SELECT {{_users}}.* FROM `_users` WHERE `email`='youtube@gmail.com' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE (`id`='profiles') OR (`name`='profiles') LIMIT 1
[0.00ms] SELECT `profiles`.* FROM `profiles` WHERE `userId`='CsySZblR0spnfZg' LIMIT 1
```
I refresh
but the data does not appear in the browser

I have filled in the user and password correctly

**Reply by ganigeorgiev** [reply]
The subscribe action only get triggered when there are changes, aka. create, update, delete operation on some of your records in the collection. Additionally the collection List and View rules also applies depending on the subscription type. This is explained in https://pocketbase.io/docs/api-realtime/, but for clarity:

>**SSE events are send for create, update and delete record operations.**
>
>You could subscribe to a **single record** or to an **entire collection**.
>
>When you subscribe to a **single record**, the collection's **ViewRule** will be used to determine whether the subscriber has access to receive the event message.
>
>When you subscribe to an **entire collection**, the collection's **ListRule** will be used to determine whether the subscriber has access to receive the event message.

**Comment by bobwatcherx** [comment]
yes work,
but can it be like firebase firestore
if in firestore the admin changes on the frontend it changes instantly without refresh :)
I've called back the testapp data record list
but it has to be refreshed so that it changes

**Comment by bobwatcherx** [comment]
yes it works. thanks very much
